src/types/UiaTextRange.hpp

```diff
@@ -264,105 +284,122 @@ namespace Microsoft::Console::Interactivity::Win32
+        static const int _compareScreenCoords(Microsoft::Console::Render::IRenderData* pData,
+                                              const ScreenInfoRow rowA,
                                               const Column colA,
                                               const ScreenInfoRow rowB,
                                               const Column colB);
 
-        static std::pair<Endpoint, Endpoint> _moveByCharacter(const int moveCount,
+        static std::pair<Endpoint, Endpoint> _moveByCharacter(Microsoft::Console::Render::IRenderData* pData,
+                                                              const int moveCount,
                                                               const MoveState moveState,
                                                               _Out_ int* const pAmountMoved);
 
-        static std::pair<Endpoint, Endpoint> _moveByCharacterForward(const int moveCount,
+        static std::pair<Endpoint, Endpoint> _moveByCharacterForward(Microsoft::Console::Render::IRenderData* pData,
+                                                                     const int moveCount,
                                                                      const MoveState moveState,
                                                                      _Out_ int* const pAmountMoved);
 
-        static std::pair<Endpoint, Endpoint> _moveByCharacterBackward(const int moveCount,
+        static std::pair<Endpoint, Endpoint> _moveByCharacterBackward(Microsoft::Console::Render::IRenderData* pData,
+                                                                      const int moveCount,
                                                                       const MoveState moveState,
                                                                       _Out_ int* const pAmountMoved);
 
-        static std::pair<Endpoint, Endpoint> _moveByLine(const int moveCount,
+        static std::pair<Endpoint, Endpoint> _moveByLine(Microsoft::Console::Render::IRenderData* pData,
+                                                         const int moveCount,
                                                          const MoveState moveState,
                                                          _Out_ int* const pAmountMoved);
 
-        static std::pair<Endpoint, Endpoint> _moveByDocument(const int moveCount,
+        static std::pair<Endpoint, Endpoint> _moveByDocument(Microsoft::Console::Render::IRenderData* pData,
+                                                             const int moveCount,
                                                              const MoveState moveState,
                                                              _Out_ int* const pAmountMoved);
 
         static std::tuple<Endpoint, Endpoint, bool>
-        _moveEndpointByUnitCharacter(const int moveCount,
+        _moveEndpointByUnitCharacter(Microsoft::Console::Render::IRenderData* pData,

```

Comment:
That's exactly what I was thinking reading this delta. If _all_ of these need to take some object as a param, then it seems to me like they should be methods, not static functions. 
This could be combined with our other note that _I'm not super comfortable with the methods added to `IRenderData`_. Maybe we need some other interface for conhost and terminal to host both those functions, and these functions here.

Reply:
I _want to_ de-`static`-fy these functions but I remember that caused a lot more trouble than it's worth. What's funny is that it propagates across a bunch of functions just like this `IRenderData` parameter did because a `static` function can only be composed of other `static` functions. By passing `IRenderData` around, we don't have to worry about changing any of the logic from other files.

I'll add it as something to investigate along with the interface architecture discussion from below (once we decide on that).

Reply:
Yeah, let's just take it as a follow-on exploration. I'm happy enough right now that this is something that is working.

Reply:
Issue #1993